### PR TITLE
Concatenate mixed static + dynamic attribute values

### LINKED ISSUE #58: string interpolation in attribute does not work as expected
```js
html`<div class="${'1'} ${'2'} ${'3'}"></div>`
```
currently returns

```
VNode {
  nodeName: 'div',
  children: [],
  attributes: { class: '1' },
  key: undefined }
```
expected
```
VNode {
  nodeName: 'div',
  children: [],
  attributes: { class: '1 2 3' },
  key: undefined }
```

### PR DESCRIPTION
**Pull request by jviide**
This pull request implements string concatenation for mixed static + dynamic attributes and adds related tests.  As long as there's more than one dynamic and/or static value they get interpreted as strings and concatenated. In practice this means that `<div a="${1}2" />`, `<div a="1${2}" />`, `<div a="${1}${2}" />`, `<div a=${1}${2} />` and so on are effectively parsed as `<div a=${"1" + "2"} />`.

As a side effect HTM now allows empty attribute names like `<div =1 />` or `<div ""=1 />`.

The `htm.mjs.br` size changes by +26 bytes (540 B -> 566 B) for the default build and +18 bytes (376 B -> 394 B) for the mini build.

Closes #58.

### CODE REVIEW

**Comment by jviide** [comment]
Marked this PR as draft, because it doesn't yet implement the string concatenation for the Babel plugins.

**Comment by jviide** [comment]
Now babel-plugin-htm also supports concatenating mixed static + dynamic attribute values. Some refactoring was necessary. The `build` and `evaluate` functions are in their own internal module now, along with a new `treeify` function. `treeify` takes the output of `build` and produces a tree-like structure that's hopefully bit easier to analyze and transform without needing to know about the gnarly size optimizations inside `build`/`evaluate`. The source code contains a small example how such a tree looks like.

The logic tries to keep the resulting code at least somewhat compact. For example `<a b=${1}${2}>` transforms to `h("a",{b:1+""+2})` because the string coercion is required. However, `<a b=${"1"}${2}>` transforms to `h("a",{b:"1"+2})`, as the `+ ""` part would be unnecessary.

**Comment by jviide** [comment]
Pushed a commit that fixed a performance regression in the non-mini version. Now the master and this PR get the same results when running `npm run test:perf`.

As a nice bonus the build size shrunk by 3 bytes, so the in total this branch adds +23 bytes to the non-mini build.

File: packages/babel-plugin-htm/index.mjs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import htm from 'htm';
+import { build,	treeify } from '../../src/build.mjs';
```
Comment (developit) [inline packages/babel-plugin-htm/index.mjs]: Does this get bundled? If not the path will fail when published.

Comment (jviide) [reply packages/babel-plugin-htm/index.mjs]: Yes, `build` and `treeify` get bundled to `dist/babel-plugin-htm.js` (and the corresponding `.mjs` file).

(Fixed the weird tab between `{ build,` and `treeify }` though 🙂)

**Pull request merged**